src/modules/townhalls/townhalls.ts

```diff
@@ -66,8 +72,46 @@ export async function getBillInfo(townhallId: string) {
     const townhall = await Collections.Townhalls().findOne({
         _id: new ObjectID(townhallId),
     });
+
     if (!townhall) throw new Error('Invalid Townhall ID');
 
-    // eventually after doing some other requests
-    return {};
+    const subject = await getSubjectUrl(townhall?.form.topic);
+    const billUrls: string[] = await getBillUrls(subject);
+    const allBills: Bill[] = [];
+    for (let i = 0; i < billUrls.length; i += 1) {
+        // eslint-disable-next-line no-await-in-loop
+        allBills.push(await getBill(billUrls[i]));
+    }
```

Comment:
I would suggest reading more about the rule here.  https://eslint.org/docs/rules/no-await-in-loop

In summary, the code here runs each of the async requests sequentially, whereas you could be running them all at the same time as shown in the link
```js
async function foo(things) {
  const results = [];
  for (const thing of things) {
    // Good: all asynchronous operations are immediately started.
    results.push(bar(thing));
  }
  // Now that all the asynchronous operations are running, here we wait until they all complete.
  return baz(await Promise.all(results));
}
```

For even more optimization, this could be converted to a map.

```js
const promises = billUrls.map((url) => getBill(url));
const allBills = await Promise.all(promises);
```

Make sure to at least glance at the rule before disabling it to make sure it's not an optimization one.  Sometimes the rules are there because you could be optimizing the code and other times it's stylistic.  You can get the link to the rule directly from vscode.